pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -102,33 +103,54 @@ func (r *RootSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 	defer r.lock.Unlock()
 
 	rsRef := req.NamespacedName
-	log := r.log.WithValues("rootsync", rsRef.String())
 	start := time.Now()
 	var err error
 	rs := &v1beta1.RootSync{}
 	reconcilerRef := types.NamespacedName{
 		Namespace: configmanagement.ControllerNamespace,
 		Name:      core.RootReconcilerName(rsRef.Name),
 	}
+	log := r.log.WithValues(
+		logFieldObject, rsRef.String(),
+		logFieldKind, r.syncKind,
+		logFieldReconciler, reconcilerRef.String())
 
 	if err = r.client.Get(ctx, rsRef, rs); err != nil {
 		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
 		if apierrors.IsNotFound(err) {
 			r.clearLastReconciled(rsRef)
-			r.log.Info("Deleting managed objects",
-				logFieldObject, rsRef.String(),
-				logFieldKind, r.syncKind)
-			return controllerruntime.Result{}, r.deleteClusterRoleBinding(ctx, reconcilerRef)
+			// TODO: Handle cleanup of RootSync deleted while reconciler-manager is down.
```

Comment:
This TODO can be handled just by removing the sync cache. I don't think we need it. Just have the reconciler-manager run deleteManagedObjects when it's triggered. It's ok if it happens multiple times, as long as it's not happening continuously.